synchronicity/type_stubs.py

```diff
@@ -781,6 +781,22 @@ def _get_function_source(
         body_indent: str,
         transform_signature=None,
     ) -> str:
+        maybe_decorators = ""
+        if hasattr(func, "__dataclass_transform__"):
+            dt_spec = func.__dataclass_transform__
+            if dt_spec["field_specifiers"]:
+                refs = ""
+                for field_spec_entity in dt_spec["field_specifiers"]:
+                    if field_spec_entity.__module__ == self.target_module:
+                        ref = field_spec_entity.__qualname__
+                    else:
+                        self.imports.add(field_spec_entity.__module__)
+                        ref = f"{field_spec_entity.__module__}.{field_spec_entity.__qualname__}"
+                    refs += ref + ", "
+
+                args = f"field_specifiers=({refs})"
+            maybe_decorators = f"{signature_indent}@typing_extensions.dataclass_transform({args})\n"
```

Comment:
typing_extensions is forward compatible so it's intentional that we always emit that version in the type stubs in order to support both older and newer python versions. I think dataclass_transform wasn't introduced in typing until 3.11 so it will be a while before we can migrate this, but type checkers support both so it shouldn't be an issue that we use typing_extensions to be safe

Reply:
That makes sense — but are we guaranteed to have `typing_extensions` imported in the type stubs we emit?